src/measure_theory/decomposition/signed_hahn.lean

```diff
@@ -0,0 +1,479 @@
+    exact id },
+  { intro x,
+    rw [set.inf_eq_inter, set.inter_comm, restrict_nonpos_seq_disjoint' h],
+    exact id }
+end
+
+private lemma exists_subset_restrict_nonpos' (hi₁ : measurable_set i) (hi₂ : s i < 0)
+  (hn : ¬ ∀ n : ℕ, ¬ s ≤[i \ ⋃ l < n, restrict_nonpos_seq s i l] 0) :
+  ∃ j : set α, measurable_set j ∧ j ⊆ i ∧ s ≤[j] 0 ∧ s j < 0 :=
+begin
+  by_cases s ≤[i] 0,
+  { exact ⟨i, hi₁, set.subset.refl _, h, hi₂⟩ },
+  { push_neg at hn,
+    set k := nat.find hn with hk₁,
+    have hk₂ : s ≤[i \ ⋃ l < k, restrict_nonpos_seq s i l] 0 := nat.find_spec hn,
+    have hmeas : measurable_set (⋃ (l : ℕ) (H : l < k), restrict_nonpos_seq s i l) :=
+      (measurable_set.Union $ λ _, measurable_set.Union_Prop
+        (λ _, restrict_nonpos_seq_measurable_set _)),
+    refine ⟨i \ ⋃ l < k, restrict_nonpos_seq s i l, hi₁.diff hmeas, set.diff_subset _ _, hk₂, _⟩,
+    rw [of_diff hmeas hi₁, s.of_disjoint_Union_nat],
+    { have h₁ : ∀ l < k, 0 ≤ s (restrict_nonpos_seq s i l),
+      { intros l hl,
+        refine le_of_lt (measure_of_restrict_nonpos_seq h _ _),
+        refine mt (restrict_le_zero_subset _ (hi₁.diff _) (set.subset.refl _)) (nat.find_min hn hl),
+        exact (measurable_set.Union $ λ _, measurable_set.Union_Prop
+          (λ _, restrict_nonpos_seq_measurable_set _)) },
+      suffices : 0 ≤ ∑' (l : ℕ), s (⋃ (H : l < k), restrict_nonpos_seq s i l),
+      { rw sub_neg,
+        exact lt_of_lt_of_le hi₂ this },
+      refine tsum_nonneg _,
+      intro l, by_cases l < k,
+      { convert h₁ _ h,
+        ext x,
+        rw [set.mem_Union, exists_prop, and_iff_right_iff_imp],
+        exact λ _, h },
+      { convert le_of_eq s.empty.symm,
+        ext, simp only [exists_prop, set.mem_empty_eq, set.mem_Union, not_and, iff_false],
+        exact λ h', false.elim (h h') } },
+    { intro, exact measurable_set.Union_Prop (λ _, restrict_nonpos_seq_measurable_set _) },
+    { intros a b hab x hx,
+      simp only [exists_prop, set.mem_Union, set.mem_inter_eq, set.inf_eq_inter] at hx,
+      exact let ⟨⟨_, hx₁⟩, _, hx₂⟩ := hx in restrict_nonpos_seq_disjoint a b hab ⟨hx₁, hx₂⟩ },
+    { apply set.Union_subset,
+      intros a x,
+      simp only [and_imp, exists_prop, set.mem_Union],
+      intros _ hx,
+      exact restrict_nonpos_seq_subset _ hx },
+    { apply_instance } }
+end .
+
+/-- A measurable set of negative measure has a negative subset of negative measure. -/
+theorem exists_subset_restrict_nonpos (hi : s i < 0) :
+  ∃ j : set α, measurable_set j ∧ j ⊆ i ∧ s ≤[j] 0 ∧ s j < 0 :=
+begin
+  have hi₁ :  measurable_set i :=
+    classical.by_contradiction (λ h, ne_of_lt hi $ s.not_measurable h),
+  by_cases s ≤[i] 0,
+  { exact ⟨i, hi₁, set.subset.refl _, h, hi⟩ },
+  { by_cases hn : ∀ n : ℕ, ¬ s ≤[i \ ⋃ l < n, restrict_nonpos_seq s i l] 0,
+    { set A := i \ ⋃ l, restrict_nonpos_seq s i l with hA,
+      set bdd : ℕ → ℕ := λ n,
+        find_exists_one_div_lt s i (⋃ k ≤ n, restrict_nonpos_seq s i k) with hbdd,
+
```

Comment:
usually, we avoid empty lines in proofs. Often, if you want to introduce a pause in an argument, to clarify what is going on, an even better way is to write a comment explaining where you are in the proof and what you will do now. Same thing below.

Reply:
I was imitating the style from this [proof](https://github.com/leanprover-community/mathlib/blob/master/src/linear_algebra/quadratic_form.lean#L758) though I see its an uncommon style. I will remove the breaks in that case